### PR TITLE
New tooltip and popover

### PR DESCRIPTION
**Pull request by mrMarshalX**
# What

- added new `ExtendedTooltip`
- added new `ExtendedPopover`

## Compatibility

- [x] Does this change maintain backward compatibility?

### Extended Tooltip
image
image
image
image

### Extended Popover
image
image
image
image
image
image

### CODE REVIEW

**Review by andrev321 — APPROVED** [review]

File: src/hooks/useHideOnScroll.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import * as React from "react";
+
+type Props = {
+  handleScroll: () => void;
+};
+
+export function useHideOnScroll({ handleScroll }: Props) {
```
Comment (nowyDEV) [inline src/hooks/useHideOnScroll.ts:7]: this should be more generic by looking at the implementation, we could also introduce a more general hook like

https://github.com/streamich/react-use/blob/master/docs/useEvent.md

and build this one on top of it

```suggestion
export function useScrollEvent({ handleScroll }: Props) {
```

File: src/hooks/useHandleClickOutside.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import * as React from "react";
+
+type Props = {
+  handleClickOutside: (e: unknown) => void;
+  clickOutsideToClose: boolean;
+};
+
+export function useHandleClickOutside({
```
Comment (nowyDEV) [inline src/hooks/useHandleClickOutside.ts:8]: I don't see the implementation of "clicking outside" here (just a mousedown event),
I think this should be implemented with the usage of ref like below

https://github.com/streamich/react-use/blob/master/docs/useClickAway.md

File: src/components/extendedPopover/ExtendedPopover.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  useHideOnScroll({ handleScroll });
+
+  if (content == null) {
+    return null;
+  }
+
+  const triggerProps = {
+    ...(triggerOn === "click"
+      ? {
+          onClick: () => setOpen(!open),
```
Comment (nowyDEV) [inline src/components/extendedPopover/ExtendedPopover.tsx:91]: better be safe and use function update
```suggestion
          onClick: () => setOpen(prevState => !prevState),
```

File: src/components/extendedPopover/ExtendedPopover.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,749 @@
...
+import { SVGIcon } from "../svgicon";
+
+<Meta title="Components/ExtendedPopover" component={ExtendedPopover} />
+
+# Extended Popover
+
+## properties
+
+> - `content`: `ReactNode` {required} popover content
+> - `trigger`: `ReactNode` {optional} popover trigger element, default to `<Button><SVGIcon name="burgerMenu" /></Button>`)
```
Comment (nowyDEV) [inline src/components/extendedPopover/ExtendedPopover.stories.mdx:12]:
```suggestion
> - `trigger`: `ReactNode` {optional} popover trigger element, defaults to `<Button><SVGIcon name="burgerMenu" /></Button>`)
```

File: src/components/extendedPopover/ExtendedPopover.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,749 @@
...
+> - `placement`: `top | bottom | right | left` {optional} relative placement of popover compared to parent element, default is `right`
+> - `position`: `start | end | center` {optional} relative shift of popover placement compared to parent element, default is `start`
+> - `margin`: `{ top?: number, bottom?: number, left?: number, right?: number }` {optional} margin around popover, default to `5px` around
```
Comment (nowyDEV) [inline src/components/extendedPopover/ExtendedPopover.stories.mdx:15]:
```suggestion
> - `margin`: `{ top?: number, bottom?: number, left?: number, right?: number }` {optional} margin around popover, defaults to `5px` for all properties
```

File: src/components/extendedTooltip/ExtendedTooltip.tsx [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+      opacity: 1;
+    }
+  }
+`;
+
+const TooltipWrapper = styled.div`
+  display: inline-block;
+`;
+
+function getColor(status: "danger" | "warning" | "success" | "notification") {
```
Comment (nowyDEV) [inline src/components/extendedTooltip/ExtendedTooltip.tsx:151]: Let's make a type for Status, we could already have it in GNUI

File: src/components/extendedTooltip/ExtendedTooltip.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,437 @@
...
+
+# Extended Tooltip
+
+## properties
+
+> - `caption`: `ReactNode` {required} tooltip content
+> - `placement`: `top | bottom | right | left` {optional} relative placement of tooltip compared to parent element, default is `top`
+> - `position`: `start | end | center` {optional} relative shift of tooltip placement compared to parent element, default is `center`
+> - `status`: `danger | warning | success | notification` {optional} background color of tooltip
+> - `margin`: `{ top?: number, bottom?: number, left?: number, right?: number }` {optional} margin around tooltip, default to `5px` around
```
Comment (nowyDEV) [inline src/components/extendedTooltip/ExtendedTooltip.stories.mdx:15]:
```suggestion
> - `margin`: `{ top?: number, bottom?: number, left?: number, right?: number }` {optional} margin around tooltip, defaults to `5px` for all properties
```

**Review by nowyDEV — COMMENTED** [review]
:clap:

File: src/hooks/useHideOnScroll.ts (hunk 1 above)
Comment (mrMarshalX) [reply src/hooks/useHideOnScroll.ts:7]: change name and implementation :wink:

File: src/hooks/useHandleClickOutside.ts (hunk 2 above)
Comment (mrMarshalX) [reply src/hooks/useHandleClickOutside.ts:8]: Hmm actually implemented it using `useEvent`, the problem with using `ref` here is that we have two `refs` to handle 😄

File: src/components/extendedPopover/ExtendedPopover.stories.mdx (hunk 4 above)
Comment (mrMarshalX) [reply src/components/extendedPopover/ExtendedPopover.stories.mdx:12]: 😱  good catch :wink:

File: src/components/extendedPopover/ExtendedPopover.stories.mdx (hunk 5 above)
Comment (mrMarshalX) [reply src/components/extendedPopover/ExtendedPopover.stories.mdx:15]: 👍

**Review by andrev321 — DISMISSED** [review]

**Review by nowyDEV — APPROVED** [review]

**Comment by Horay** [comment]
Screenshot 2022-01-19 at 10 33 09
here the text color should be "theme.color.text.text04"

**Comment by mrMarshalX** [comment]
Seems colors are working fine, verified with @Horay 😏

File: src/components/extendedPopover/ExtendedPopover.tsx [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+      }
+    },
+    [contentRef, triggerRef, clickOutsideToClose]
+  );
+
+  useEvent({ name: "click", handler: handleClickOutside, target: document });
+
+  const handleScroll = throttle(
+    React.useCallback(() => {
+      console.log("asd");
```
Comment (nowyDEV) [inline src/components/extendedPopover/ExtendedPopover.tsx:82]: fdsa :D

**Review by nowyDEV — APPROVED** [review]

**Review by Bosmanfrx — APPROVED** [review]

**Review by andrev321 — APPROVED** [review]

**Pull request merged**